jenkins/scripts/integration_delete.sh

```diff
@@ -17,8 +17,21 @@ source "${CI_DIR}/utils.sh"
 
 TEST_EXECUTER_PORT_NAME="${TEST_EXECUTER_PORT_NAME:-${TEST_EXECUTER_VM_NAME}-int-port}"
 
-# Delete executer vm
+# Delete executer VM
 echo "Deleting executer VM."
 openstack server delete "${TEST_EXECUTER_VM_NAME}"
 
-openstack port delete "${TEST_EXECUTER_PORT_NAME}"
+echo "Waiting until volume status is available, to proceed with proper volume deletion."
+until openstack volume show "${TEST_EXECUTER_VM_NAME}" -f json \
+  | jq .status | grep "available"
+do
+  sleep 10
+done
```

Comment:
you should maybe add an exit condition here. If the status is error, it will probably not get out of that on its own, and you should proceed with deletion.